### PR TITLE
Add dynamic timeout feature

### PR DESCRIPTION
**Pull request by saguziel**
This adds a feature to set the Distcp timeout to scale with the file input size, which can subject to a minimum (because of distcp overhead) and a maximum( to resemble older behavior).

@plypaul

### CODE REVIEW

File: main/src/main/java/com/airbnb/reair/incremental/DirectoryCopier.java [hunk 1]
```diff
@@ -80,7 +83,24 @@ public long copy(Path srcDir, Path destDir, List<String> copyAttributes) throws
           ConfigurationKeys.COPY_JOB_TIMEOUT_SECONDS,
           -1);
       if (copyJobTimeoutSeconds > 0) {
-        options.setDistCpJobTimeout(copyJobTimeoutSeconds * 1000);
+        options.setDistCpJobTimeout(copyJobTimeoutSeconds * 1_000L);
+      }
+      boolean dynamicTimeoutEnabled = conf.getBoolean(ConfigurationKeys.COPY_JOB_DYNAMIC_TIMEOUT_ENABLED, false);
+      options.setDistcpDynamicJobTimeoutEnabled(dynamicTimeoutEnabled);
+      if (dynamicTimeoutEnabled && copyJobTimeoutSeconds > 0) {
+        LOG.warn("Trying to set static timeout and enable dynamic timeout for DistCp jobs at the same time.");
```
Comment (plypaul) [inline main/src/main/java/com/airbnb/reair/incremental/DirectoryCopier.java]: Throw appropriate configuration exception here?

**Comment by plypaul** [comment]
@saguziel Please take a look at the build failure.

**Comment by saguziel** [comment]
@plypaul  ptal

**Comment by plypaul** [comment]
Thanks @saguziel ! Merging now.

**Pull request merged**